From 8984ba33d18fafdb1f1fe0b833a97f275cf41ff3 Mon Sep 17 00:00:00 2001
From: Venkateshwarlu Domakonda <Venkateshwarlu@codeaurora.org>
Date: Thu, 22 Nov 2012 19:35:13 +0530
Subject: [PATCH] qcom-fm: Add proper permission for FM to get/set property
 variables

Allow FM Process to set and get the system property variables
specific for FM operation.

For mr1 based qcom fm solution.

Change-Id: Idc937e5c3ca4fbb9fbfe2e4e9eed9eb9bd50499c
---
 include/private/android_filesystem_config.h |    6 ++++++
 init/property_service.c                     |    6 ++++++
 2 files changed, 12 insertions(+)

diff --git a/include/private/android_filesystem_config.h b/include/private/android_filesystem_config.h
index 94b26cf..eda4085 100644
--- a/include/private/android_filesystem_config.h
+++ b/include/private/android_filesystem_config.h
@@ -73,6 +73,9 @@
 #define AID_LOOP_RADIO    1030  /* loop radio devices */
 #define AID_MEDIA_DRM     1031  /* MediaDrm plugins */
 #define AID_AUDIT         1032  /* audit daemon */
+#ifdef QCOM_FM_ENABLED
+#define AID_FM_RADIO      1033  /* FM radio */
+#endif
 
 #define AID_THEMEMAN      1300  /* theme manager */
 
@@ -187,6 +190,9 @@ static const struct android_id_info android_ids[] = {
     { "mot_dlna",  AID_MOT_DLNA, },
 #endif
     { "misc",      AID_MISC, },
+#ifdef QCOM_FM_ENABLED
+    { "fm_radio",  AID_FM_RADIO, },
+#endif
     { "nobody",    AID_NOBODY, },
     { "clat",      AID_CLAT, },
     { "mediadrm",  AID_MEDIA_DRM, },
diff --git a/init/property_service.c b/init/property_service.c
index c4cb1e5..ecab5f7 100755
--- a/init/property_service.c
+++ b/init/property_service.c
@@ -98,6 +98,9 @@ struct {
     { "persist.security.", AID_SYSTEM,   0 },
     { "persist.service.bdroid.", AID_BLUETOOTH,   0 },
     { "selinux."         , AID_SYSTEM,   0 },
+#ifdef QCOM_FM_ENABLED
+    { "hw.fm.",           AID_FM_RADIO,  0 },
+#endif
     { "net.pdp",          AID_RADIO,    AID_RADIO },
     { "service.bootanim.exit", AID_GRAPHICS, 0 },
 #ifdef PROPERTY_PERMS_APPEND
@@ -120,6 +123,9 @@ struct {
 } control_perms[] = {
     { "dumpstate",AID_SHELL, AID_LOG },
     { "ril-daemon",AID_RADIO, AID_RADIO },
+#ifdef QCOM_FM_ENABLED
+    { "fm_dl", AID_FM_RADIO, AID_FM_RADIO},
+#endif
 #ifdef CONTROL_PERMS_APPEND
 CONTROL_PERMS_APPEND
 #endif
-- 
1.7.9.5

